bugbot/rules/uplift_beta.py

```diff
@@ -82,6 +82,34 @@ def bug_handler(bug, data):
 
         return bugs_without_regr
 
+    def filter_bugs_by_ni(self, bugs):
+        bugs_without_ni_on_assignee = {}
+
+        def bug_handler(bug, data):
+            bugid = str(bug["id"])
+
+            needinfos = [
+                {"requestee": flag["requestee"], "type": flag["name"]}
+                for flag in bug.get("flags", [])
+                if flag["name"] == "needinfo"
+            ]
+
+            assignee = bug.get("assigned_to", "Unassigned")
+
+            ni_for_assignee = any(ni["requestee"] == assignee for ni in needinfos)
+
+            if not ni_for_assignee:
+                data[bugid] = bugs[bugid]
+
+        Bugzilla(
+            bugids=list(bugs.keys()),
+            include_fields=["id", "flags", "assigned_to"],
+            bughandler=bug_handler,
+            bugdata=bugs_without_ni_on_assignee,
+        ).get_data().wait()
```

Comment:
Why not add the missing field to the primary query in `get_bz_params()` instead of doing an additional query?

Reply:
I'm attempting to tweak the Bugzilla search parameters but I'm having some difficulty finding the right ones. I've tested the following parameters but I've gotten different results compared to the current implemented filters:
```
            "f8": "OP",
            "f9": "flagtypes.name",
            "o9": "substring",
            "v9": "needinfo",
            "f10": "flagtypes.requestee",
            "o10": "notequals",
            "v10": "%assignee%",
            "f11": "CP"
```

Reply:
20b46bb14a84fe156812efefbf7f35e16957bf6d